packages/api-2/src/index.ts

```diff
@@ -0,0 +1,114 @@
+// Copyright 2017-2018 @polkadot/api authors & contributors
+// This software may be modified and distributed under the terms
+// of the ISC license. See the LICENSE file for details.
+
+import { QueryableStorageFunction, QueryableModuleStorage, QueryableStorage, SubmittableExtrinsics, SubmittableModuleExtrinsics, SubmittableExtrinsicFunction } from './types';
+
+import { Observable } from 'rxjs';
+import WsProvider from '@polkadot/api-provider/ws';
+import defaults from '@polkadot/api-rx/defaults';
+import RpcRx from '@polkadot/api-rx/index';
+import { Extrinsics } from '@polkadot/extrinsics/types';
+import extrinsicsFromMeta from '@polkadot/extrinsics/fromMetadata';
+import { Storage } from '@polkadot/storage/types';
+import storageFromMeta from '@polkadot/storage/fromMetadata';
+import { Method } from '@polkadot/types/index';
+import logger from '@polkadot/util/logger';
+
+import SubmittableExtrinsic from './SubmittableExtrinsic';
+
+const l = logger('api-rx');
+
+export default class ApiRx {
+  // @ts-ignore Indirectly decorated inside the constructor
+  private _extrinsics: SubmittableExtrinsics;
+  // @ts-ignore Indirectly decorated inside the constructor
+  private _state: QueryableStorage;
+  private _rpc: RpcRx;
+
+  constructor (wsProvider: WsProvider = new WsProvider(defaults.WS_URL)) {
+    this._rpc = new RpcRx(wsProvider);
+
+    // !#$%@ But yes, handling promises here
+    this.initMetadata().then().catch();
+  }
+
+  get extrinsics (): SubmittableExtrinsics {
+    return this._extrinsics;
+  }
+
+  get rpc (): RpcRx {
+    return this._rpc;
+  }
+
+  get state (): QueryableStorage {
```

Comment:
Basically -

```
// returns observable that will change when the value changes
api.state.balances.freeBalance(<my-address>) 
```

Reply:
Or possibly `api.query.balancse...`

Reply:
`api.storage` would be the most intuitive, too long?